packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol

```diff
@@ -1173,46 +997,61 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
         // Only freeze the round if this is the first request in the round.
         if (rounds[roundId].gat == 0) {
             // Set the round gat percentage to the current global gat rate.
+
             rounds[roundId].gat = gat;
 
             // Store the cumulativeStake at this roundId to work out slashing and voting trackers.
             rounds[roundId].cumulativeStakeAtRound = cumulativeStake;
         }
     }
 
-    // Returns if a given price request, with known votingInstance and currentRoundId is resolved.
-    function _priceRequestResolved(
-        PriceRequest storage priceRequest,
-        VoteInstance storage voteInstance,
-        uint256 currentRoundId
-    ) private returns (bool) {
-        // We are currently either in the voting round for the request or voting is yet to begin.
-        if (currentRoundId <= priceRequest.lastVotingRound) return false;
-
-        // If the request has been previously resolved, return true.
-        if (priceRequest.pendingRequestIndex == UINT64_MAX) return true;
-
-        // Else, check if the price can be resolved.
-        (bool isResolvable, int256 resolvedPrice) =
-            voteInstance.resultComputation.getResolvedPrice(_computeGat(priceRequest.lastVotingRound));
-
-        // If it's not resolvable return false.
-        if (!isResolvable) return false;
-
-        // Else, the request is resolvable. Remove the element from the pending requests and update pendingRequestIndex
-        // within the price request struct to make the next entry into this method a no-op for this request.
-        _removeRequestFromPendingPriceRequests(priceRequest.pendingRequestIndex);
-
-        priceRequest.pendingRequestIndex = UINT64_MAX;
-        emit PriceResolved(
-            priceRequest.lastVotingRound,
-            priceRequest.priceRequestIndex,
-            priceRequest.identifier,
-            priceRequest.time,
-            priceRequest.ancillaryData,
-            resolvedPrice
-        );
-        return true;
+    function _resolveResolvablePriceRequests(uint256 maxTraversals) private {
+        uint32 currentRoundId = uint32(getCurrentRoundId());
+
+        uint256 index = rounds[currentRoundId].resolvableIndex;
```

Comment:
Quick question on gas: do you think we could check and delete the value from round n - 1 if the index is 0? This might give us a bit of a gas rebate when moving to the next round.

I don't think this index is needed once the round is passed, right? So it would be ideal if we could either remove it or reuse the storage slot (but this might be more annoying as we'd have to track when the round changed).

Reply:
this is a good idea but I'm going to leave it as an optimization for a later PR. this will be a marginal saving for exactly one user per round.